hs_composite_resource/models.py

```diff
@@ -237,17 +255,37 @@ def _recreate_xml_docs_for_folder(self, folder, check_metadata_dirty=False):
                 aggregation.create_aggregation_xml_documents()
 
     def get_aggregation_by_name(self, name):
-            """Get an aggregation that matches the aggregation name specified by *name*
-            :param  name: name (aggregation path) of the aggregation to find
-            :return an aggregation object if found
-            :raises ObjectDoesNotExist if no matching aggregation is found
-            """
-            for aggregation in self.logical_files:
-                # remove the last slash in aggregation_name if any
-                if aggregation.aggregation_name.rstrip('/') == name:
+        """Get an aggregation that matches the aggregation name specified by *name*
+        :param  name: name (aggregation path) of the aggregation to find
+        :return an aggregation object if found
+        :raises ObjectDoesNotExist if no matching aggregation is found
+        """
+        for aggregation in self.logical_files:
+            # remove the last slash in aggregation_name if any
+            if aggregation.aggregation_name.rstrip('/') == name:
+                return aggregation
+
+        raise ObjectDoesNotExist("No matching aggregation was found for name:{}".format(name))
+
+    def get_fileset_aggregation_in_path(self, path):
+        """Get the first fileset aggregation in the path moving up (towards the root)in the path
+        :param  path: directory path in which to search for a fileset aggregation
+        :return a fileset aggregation object if found, otherwise None
+        """
+        while True:
```

Comment:
what?  This took me a minute to understand, I thought for sure it was left over some from some debugging or something.

How about:
```
while '/' in path:
    ...
    path = os.path.dirname(path)
```